### ISSUE TITLE
Anomaly in RPN Loss function

### ISSUE THREAD

**Issue by Abhijit-2592** [yhenon/keras-frcnn#225]
The rpn classification loss function doesn't seem to match the original implementation (Faster-rcnn paper) nor the tensorflow's object detection api paper (speed and accuracy trade off ... paper)

@yhenon  could you please explain your implementation of RPN_classification loss function?
From the original paper it should look something like 

`binary_crossentropy(y_pred[:, :, :, :], y_true[:, :, :, self.num_anchors:]) / self.N_cls # normalization`

But you have implemented it as:
`return lambda_rpn_class * K.sum(y_true[:, :, :, :num_anchors] * K.binary_crossentropy(y_pred[:, :, :, :], y_true[:, :, :, num_anchors:])) / K.sum(epsilon + y_true[:, :, :, :num_anchors])`

more over inside the binary_cross entropy it is given as `y_true[:,:,:,num_anchors:]`, shouldn't it be `y_true[:,:,:,:num_anchors]` ?

I don't seem to understand this. Could you please give some hints?

**Comment by thealchemist89** [comment]
@Abhijit-2592, Do you have any clues on your above question?

**Comment by Abhijit-2592** [comment]
@thealchemist89  Nope
